frontend/packages/dev-console/src/components/pipelines/pipeline-form/StartPipelineForm.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Form } from '@patternfly/react-core';
+import { FormikValues } from 'formik';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import PipelineResourceSection, { ResourceProps } from './PipelineResourceSection';
+import PipelineParameterSection from './PipelineParameterSection';
+
+const StartPipelineForm: React.FC<FormikValues> = ({
+  values,
+  errors,
+  handleSubmit,
+  status,
+  isSubmitting,
+  close,
+}) => {
+  const resources: ResourceProps = values.resources.reduce(
+    (acc, value, index) => {
+      const resource = _.cloneDeep(value);
```

Comment:
> `cloneDeep` can be avoided if instead of a `merge` it does `{ ...resource, index }`.

OOOOoh right, Lodash.merge mutates the first object. Forgot about that.